File: web-development/level-1/helsinki-part-1-2/Java/part1/week5/week05-ex85-Lyrra/LyrraCard.java
public class LyrraCard  {
  //fields
  private double balance;

  //consturctors
  public LyrraCard(double balance)  {
    this.balance = balance;
  }

  //behaviors
  public double getBalance()  {
    return this.balance;
  }

  public void loadMoney(double amount) {
    this.balance += amount;
  }

  public boolean pay(double amount) {
    if (this.balance < amount) {
      return false;
    }
    this.balance -= amount;
    return true;
  }

//class ends here
}

File: helsinki-moocs/python/week05/ex86/LyrraCard.py
class LyrraCard:

    #constructor
    def __init__(self, balance):
        self.balance = balance

    #behaviors

    #gets the balance
    def balance(self):
        return self.balance

    #loads money
    def load_money(self, amount):
        self.balance += amount

    #paying system
    def pay(self, amount):
        if self.balance > amount and self.balance > 0:
            self.balance -= amount
        else:
            print('Not enough balance : ', self.balance)
